### PR TITLE
Further optimizations to VerifyWithBuffer

### PR DESCRIPTION
**Pull request by boreq**
Environment:

```
goos: linux
goarch: amd64
cpu: AMD Ryzen 7 3700X 8-Core Processor 
```

Results from `master`:
```
BenchmarkVerify-16    	     633	  18817784 ns/op	 5285065 B/op	   23057 allocs/op
```
Results from this pull request:

```
BenchmarkVerify-16    	     694	  16936649 ns/op	 2035855 B/op	   17710 allocs/op
```

This pull request further reduces the execution time to 90% of what it is on `master` and number of allocs to 77% of what it is on `master`.

### CODE REVIEW

File: message/legacy/replace.go [hunk 1]
```diff
@@ -74,27 +73,33 @@ func quoteString(buf *bytes.Buffer, s string) {
...
+
 // InternalV8Binary does some funky v8 magic
 // new Buffer(in, "binary") returns soemthing like (u16 && 0xff)
 func InternalV8Binary(in []byte) ([]byte, error) {
-	var u16 bytes.Buffer
-	enc := unicode.UTF16(unicode.LittleEndian, unicode.IgnoreBOM).NewEncoder()
-	trans := transform.NewWriter(&u16, enc)
-	if _, err := io.Copy(trans, bytes.NewReader(in)); err != nil {
-		return nil, fmt.Errorf("internalV8bin: failed to transform input to u16: %w", err)
+	guessedLength := len(in) * 2
```
Comment (boreq) [inline message/legacy/replace.go:81]: If someone knows the exact formula please let me know. I think it is `1.5` going off the top of my head but I may be completely wrong. It is fine to go overboard here as it won't cause any real performance problems (we just care that we are doing fewer allocs not how large they are) but I want to make sure that we are not erroring out on weird input.

**Comment by boreq** [comment]
It is getting harder to optimize this as at this point 81% of the execution time is something we can do nothing about; signature verification:



This is the bit that we can still optimize if anyone is curious:

**Comment by decentral1se** [comment]
Oh shiet, a failing test :sob: 

https://github.com/ssbc/go-ssb/actions/runs/4365276672/jobs/7633680649

```
 --- FAIL: TestInvalidFuzzed (0.00s)
    verify_invalid_test.go:85: 
        	Error Trace:	/home/runner/work/go-ssb/go-ssb/message/legacy/verify_invalid_test.go:85
        	Error:      	An error is expected but got nil.
        	Test:       	TestInvalidFuzzed
        	Messages:   	in msg 49
    verify_invalid_test.go:86: expected error: Message must have a valid order
    verify_invalid_test.go:85: 
        	Error Trace:	/home/runner/work/go-ssb/go-ssb/message/legacy/verify_invalid_test.go:85
        	Error:      	An error is expected but got nil.
        	Test:       	TestInvalidFuzzed
        	Messages:   	in msg 50
    verify_invalid_test.go:86: expected error: Message must have a valid order
FAIL
FAIL	github.com/ssbc/go-ssb/message/legacy	0.318s
FAIL
make: *** [Makefile:23: test] Error 1
```

**Comment by boreq** [comment]
I will check this.

**Comment by boreq** [comment]
Oh no that was my broken code. [7e77d30](https://github.com/ssbc/go-ssb/pull/334/commits/7e77d308e329facf087b97aad10cfc4bf3f3d013)

**Pull request merged**